src/blockchaintree.rs

```diff
@@ -788,10 +792,10 @@ impl BlockChainTree {
         Ok(())
     }
 
-    pub fn get_derivative_chain(
+    pub async fn get_derivative_chain(
         &mut self,
         addr: &[u8; 33],
-    ) -> Result<Option<Box<DerivativeChain>>, BlockChainTreeError> {
+    ) -> Result<Option<&Arc<RwLock<DerivativeChain>>>, BlockChainTreeError> {
```

Comment:
Should be Result<Option<Arc<RwLock<DerivativeChain>>>, BlockChainTreeError>
Arc should be cloned for response

Reply:
i can't do that
![image](https://user-images.githubusercontent.com/87724011/212565496-315fbbdc-afe1-4d75-b70c-d80ae455829d.png)

Reply:
maybe  Result<Option<Arc<RwLock<DerivativeChain>>>  thats beter

Reply:
> Should be `Result<Option<Arc<RwLock<DerivativeChain>>>`, BlockChainTreeError> Arc should be cloned for response

I fixed markdown in this line